frontend/src/components/modals/NotifyUsersModal.tsx

```diff
@@ -0,0 +1,225 @@
+    [],
+  );
+  const [errorMessage, setErrorMessage] = useState('');
+  const [allChecked, setAllChecked] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (getType(userInfo?.roles, task.roadmapId) === RoleType.Admin) {
+      const ratings = task.ratings.map((rating) => ({
+        createdByUser: rating.createdByUser,
+        forCustomer: rating.forCustomer,
+      }));
+      const missingCustomers = customers?.filter((customer) => {
+        const representativeIds = customer?.representatives?.map(
+          (rep) => rep.id,
+        );
+        return !representativeIds?.every((rep) => {
+          return ratings.some((rating) => {
+            if (
+              rating.createdByUser === rep &&
+              rating.forCustomer === customer.id
+            )
+              return true;
+            return false;
+          });
+        });
+      });
+
+      const missingRepresentatives: CheckableUser[] = [];
+
+      // Find representatives who haven't given ratings
+      if (missingCustomers) {
+        missingCustomers.forEach((customer) => {
+          // Filter out logged in user, so they won't send email to themselves
+          const representatives = customer?.representatives?.filter(
+            (user) => user.id !== userInfo?.id,
+          );
+          const ratingsForTask = task.ratings
+            .filter((rating) => rating.forCustomer === customer.id)
+            .map((e) => e.createdByUser);
+
+          if (representatives) {
+            representatives.forEach((rep) => {
+              if (!ratingsForTask.includes(rep.id))
+                if (!missingRepresentatives.some((e) => e.id === rep.id))
+                  missingRepresentatives.push({ ...rep, checked: false });
+            });
+          }
+        });
+      }
+
+      // Filter developers, check who haven't given ratings, add checked attribute for all of them
+      const ratingIds = task.ratings.map((rating) => rating.createdByUser);
+      const missingDevelopers = allUsers
+        ?.filter((user) => user.type === RoleType.Developer)
+        .filter((developer) => !ratingIds.includes(developer.id))
+        .map((dev) => ({ ...dev, checked: false }));
+
+      setMissingUsers([
+        ...(missingRepresentatives ?? []),
+        ...(missingDevelopers ?? []),
+      ]);
+    }
+  }, [allUsers, customers, task.ratings, task.roadmapId, userInfo]);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsLoading(true);
+
+    const checkedUsers = missingUsers?.filter((user) => user.checked);
+    // Parse RoadmapUsers from list of CheckedUsers
+    const usersToNotify = allUsers?.filter((user) =>
+      checkedUsers?.some((checkedUser) => checkedUser.id === user.id),
+    );
+
+    if (!usersToNotify || usersToNotify?.length === 0) {
+      setIsLoading(false);
+      return setErrorMessage(t('Please select at least one user'));
```

Comment:
Instead of showing an error when no users are selected, the submit button could be disabled.